File: python/split_into_chunks.py
#!/usr/bin/env python3
import sys

def split_into_chunks(file_path, chunk_size=5):
    with open(file_path, 'r', encoding='utf-8') as file:
        text = file.read()

    # Split the text into chunks based on multiple consecutive empty lines
    chunks = [chunk.strip() for chunk in text.strip().split('\n\n') if chunk.strip()]

    # Write chunks to files
    for i in range(0, len(chunks), chunk_size):
        file_path_no_ext = file_path.split('.')[0]
        chunk_file_path = f"{file_path_no_ext}-chunk-{i // chunk_size + 1}.txt"
        with open(chunk_file_path, 'w', encoding='utf-8') as chunk_file:
            for chunk in chunks[i:i+chunk_size]:
                chunk_file.write(chunk + '\n\n')

    return chunks

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python split_into_chunks.py <file_path>")
        sys.exit(1)

    file_path = sys.argv[1]
    split_into_chunks(file_path)

File: src/splitIntoChunks.ts
import * as fs from 'fs';

// Split a text file into smaller chunks and return the chunk file paths
export const splitIntoChunks = (filePath: string, chunkSize: number = 5): string[] => {
    const text = fs.readFileSync(filePath, 'utf-8');
    const chunks = text.trim().split('\n\n').filter(chunk => chunk.trim());
    const chunkFilePaths: string[] = [];

    for (let i = 0; i < chunks.length; i += chunkSize) {
        const filePathNoExt = filePath.split('.')[0];
        const chunkFilePath = `${filePathNoExt}-chunk-${Math.floor(i / chunkSize) + 1}.txt`;
        chunkFilePaths.push(chunkFilePath);
        const chunkContent = chunks.slice(i, i + chunkSize).join('\n\n');
        fs.writeFileSync(chunkFilePath, chunkContent + '\n\n');
    }

    return chunkFilePaths;
}